src/gpuhunt/providers/vultr.py

```diff
@@ -0,0 +1,205 @@
+        # gpu_count=1
+        return RawCatalogItem(
+            instance_name=plan.get("id"),
+            location=location,
+            price=plan.get("hourly_cost", 0),
+            cpu=plan.get("vcpu_count"),
+            memory=plan.get("ram", 0) / 1024,
+            gpu_count=gpu_count,
+            gpu_name=gpu_name,
+            gpu_memory=gpu_memory_gb,
+            gpu_vendor=gpu_vendor,
+            spot=False,
+            disk_size=plan.get("disk", 0),
+        )
+
+
+def get_gpu_memory(gpu_name: str) -> float:
+    for gpu in KNOWN_NVIDIA_GPUS:
+        if gpu.name.upper() == gpu_name:
+            return gpu.memory
+    for gpu in KNOWN_AMD_GPUS:
+        if gpu.name.upper() == gpu_name:
+            return gpu.memory
+    logger.error(f"GPU {gpu_name} not found in known GPU lists.")
+    raise ValueError(f"GPU {gpu_name} not found.")
+
+
+def get_gpu_vendor(gpu_name: Optional[str]) -> Optional[str]:
+    if gpu_name is None:
+        return None
+    for gpu in KNOWN_NVIDIA_GPUS:
+        if gpu.name.upper() == gpu_name:
+            return AcceleratorVendor.NVIDIA.value
+    for gpu in KNOWN_AMD_GPUS:
+        if gpu.name.upper() == gpu_name:
+            return AcceleratorVendor.AMD.value
+    return None
+
+
+def extract_gpu_info_from_id(id_str: str):
+    parts = id_str.split("-")
+    if "gpu" in parts:
+        gpu_name = parts[-2].upper()
+        try:
+            gpu_count = int(parts[-3])
+        except ValueError:
+            gpu_count = 1  # Default set to 1 if count is not explicitly specified,
+            # for instance in vbm-64c-2048gb-l40-gpu count is not specified but
+            # in vbm-64c-2048gb-8-l40-gpu count is specified as 8
+        return gpu_name, gpu_count
+    return None, 0
+
+
+def _make_request(method: str, path: str, data: Any = None) -> Response:
+    try:
+        response = requests.request(
+            method=method,
+            url=API_URL + path,
+            json=data,
+            timeout=30,
+        )
+        response.raise_for_status()
+        return response
+
+    except requests.HTTPError as e:
+        status_code = e.response.status_code if e.response else None
+        if status_code == requests.codes.not_found:
+            logger.exception(f"Resource not found at {API_URL + path}.")
+        elif status_code == requests.codes.bad_request:
+            logger.exception(
+                f"Bad request to {API_URL + path}. Check the request payload or parameters."
+            )
+        elif status_code == requests.codes.forbidden:
+            logger.exception(f"Access forbidden to {API_URL + path}. Check API permissions.")
+        elif status_code == requests.codes.unauthorized:
+            logger.exception(
+                f"Unauthorized access to {API_URL + path}. Check API key or authentication details."
+            )
+        else:
+            logger.exception(f"HTTP error {status_code} occurred when accessing {API_URL + path}.")
```

Comment:
I don't quite understand these messages.
> Resource not found at {API_URL + path}

The only resources we are requesting are `/plans` and `/plans-metal`, which I suppose will always be found.

> Check API key or authentication details

Our requests don't use authentication, there isn't anything to check.

In any case, I think all this error handling is safe to remove as I mentioned in a previous comment.

Reply:
Ok. What I understood is below is enough and no need for local error handling.

```
def _make_request(method: str, path: str, data: Any = None) -> Response:
    response = requests.request(
        method=method,
        url=API_URL + path,
        json=data,
        timeout=30,
    )
    response.raise_for_status()
    return response
```